perf/benchmark/run_benchmark_job.sh

```diff
@@ -189,9 +189,8 @@ pushd "${WD}"
 export ISTIO_INJECT="true"
 ./setup_test.sh
 popd
-dt=$(date +'%Y%m%d-%H')
-SHA=$(git rev-parse --short "${GIT_SHA}")
-export OUTPUT_DIR="benchmark_data-${GIT_BRANCH}.${dt}.${SHA}"
```

Comment:
(1) remove `-%H`, since we do not need time granularity to be that much fine, as long as there is date is okay
(2) the gcs bucket link: https://gcsweb.istio.io/gcs/istio-build/perf/ is already has perf meaning, we may do not need `benchmark_data` as the part of repo name in order to keep the full path short.